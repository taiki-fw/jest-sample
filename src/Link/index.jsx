import React, { useState } from "react";

const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal"
};

export default function Link({ page, children }) {
  const [status, setStatus] = useState(STATUS.NORMAL);
  const mouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };
  const mouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };
  return (
    <a
      className={status}
      href={page || "#"}
      onMouseEnter={mouseEnter}
      onMouseLeave={mouseLeave}
    >
      {children}
    </a>
  );
}
