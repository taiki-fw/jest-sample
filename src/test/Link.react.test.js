import React from "react";
import Link from "../Link.react";
import renderer, { act } from "react-test-renderer";

test("Link changes the class when hoverd", () => {
  const component = renderer.create(
    <Link page="http://www.facebook.com">facebook</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onMouseEnter();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onMouseLeave();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onClick();
  });
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
