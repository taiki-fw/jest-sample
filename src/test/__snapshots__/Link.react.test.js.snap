// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hoverd 1`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;

exports[`Link changes the class when hoverd 2`] = `
<a
  className="hovered"
  href="http://www.facebook.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;

exports[`Link changes the class when hoverd 3`] = `
<a
  className="normal"
  href="http://www.facebook.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;

exports[`Link changes the class when hoverd 4`] = `
<a
  className="clicked"
  href="http://www.facebook.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  facebook
</a>
`;
